src/device-state/preload.ts

```diff
@@ -8,21 +9,21 @@ import * as deviceConfig from '../device-config';
 import * as eventTracker from '../event-tracker';
 import * as images from '../compose/images';
 
-import constants = require('../lib/constants');
 import { AppsJsonParseError, EISDIR, ENOENT } from '../lib/errors';
 import log from '../lib/supervisor-console';
 
 import { convertLegacyAppsJson } from '../lib/migration';
 import { AppsJsonFormat } from '../types/state';
+import * as fsUtils from '../lib/fs-utils';
 
-export async function loadTargetFromFile(
-	appsPath: Nullable<string>,
-): Promise<void> {
-	log.info('Attempting to load any preloaded applications');
-	if (!appsPath) {
-		appsPath = constants.appsJsonPath;
-	}
+export function appsJsonBackup(appsPath: string) {
+	const [dirname, basename] = [path.dirname(appsPath), path.basename(appsPath)];
+
+	return path.join(dirname, `preloaded-${basename}`);
```

Comment:
Consider adding extension to file instead of changing the name. The logic is because when we make backups we do ${file}.backup. This doesn't really matter though.

```suggestion
	return path.join(dirname, `${basename}.preloaded`);
```

Reply:
The only reason I did it this way was to keep the correct extension in case you want to read the file, but that may not be necessary really.

Reply:
Do we expect people to interact with this file ? Otherwise, a "file extension" isn't meaningful here like on other systems such as Windows where it the OS will behave differently based on the extension.

Reply:
I would expect to be useful for us to read it while on support. I was under the impression that `vi` on device would use the extension for syntax highlighting but I tested and it doesn't do syntax highlighting at all. 

I'll make the change

Reply:
that's a good point my bad!

Reply:
for me the .preloaded does prevent my vim (not vi) from identifying the filetype as JS